src/main/kotlin/com/cosmotech/api/utils/KubernetesClient.kt

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+package com.cosmotech.api.utils
+
+import io.kubernetes.client.openapi.ApiException
+import io.kubernetes.client.openapi.apis.CoreV1Api
+import io.kubernetes.client.openapi.models.V1ObjectMeta
+import io.kubernetes.client.openapi.models.V1Secret
+import io.kubernetes.client.util.Config
+import java.util.Base64
+import javax.annotation.PostConstruct
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+
+@Service
+class KubernetesClient {
+
+  private val logger = LoggerFactory.getLogger(KubernetesClient::class.java)
+  private lateinit var kubernetesApi: CoreV1Api
+
+  @PostConstruct
+  internal fun init() {
+    kubernetesApi = CoreV1Api(Config.defaultClient())
+  }
+
+  fun getSecretFromKubernetes(
+      secretName: String,
+      namespace: String = "phoenix"
+  ): Pair<String, String> {
+    var name = String()
+    var key = String()
+    try {
+      val result = kubernetesApi.readNamespacedSecret(secretName, namespace, "")
+
+      result.data?.forEach { entry ->
+        run {
+          name = entry.key
+          key = String(Base64.getDecoder().decode(entry.value))
+        }
+      }
+
+      logger.info("Secret retrieved {}", result)
+    } catch (e: ApiException) {
+      logger.info("Exception when getting secret {}", e.message)
+    }
+
+    return Pair(name, key)
```

Comment:
I don't understand how you handle multiple entries in secret? Maybe add the entry name in function and in callers in api

Reply:
Can be enough in step 1

Reply:
Multiple entries is handled with the secretName
"  fun getSecretFromKubernetes(
      secretName: String,
      namespace: String = "phoenix"
  ): Pair<String, String> {"